database/src/database.rs

```diff
@@ -723,4 +723,61 @@ mod tests {
         let row = db.read(tx4, 1).await.unwrap().unwrap();
         assert_eq!(tx2_row, row);
     }
+
+    // Test for the visibility to check if a new transaction can see old committed values.
+    // This test checks for the typo present in the paper, explained in https://github.com/penberg/mvcc-rs/issues/15
+    #[traced_test]
+    #[tokio::test]
+    async fn test_committed_visibility() {
+        let clock = LocalClock::default();
+        let db = Database::<LocalClock, tokio::sync::Mutex<_>>::new(clock);
+
+        // let's add $10 to my account since I like money
+        let tx1 = db.begin_tx().await;
+        let tx1_row = Row {
+            id: 1,
+            data: "10".to_string(),
+        };
+        db.insert(tx1, tx1_row.clone()).await.unwrap();
+        db.commit_tx(tx1).await.unwrap();
+
+        // but I like more money, so let me try adding $10 more
+        let tx2 = db.begin_tx().await;
+        let tx2_row = Row {
+            id: 1,
+            data: "20".to_string(),
+        };
+        assert!(db.update(tx2, tx2_row.clone()).await.unwrap());
```

Comment:
Can you add an assertion here to check that `read(tx2, 1)` returns `tx2_row`?

Reply:
@penberg I have added this assert